src/mesh/BoundingBox.cpp

```diff
@@ -108,58 +110,58 @@ int BoundingBox::getDimension() const
 
 const std::vector<double> BoundingBox::dataVector() const
 {
-  return _bounds;
+  std::vector<double> bounds({_boundMin[0], _boundMax[0], _boundMin[1], _boundMax[1]});
+  if (_dimensions == 3) {
+    bounds.insert(bounds.end(), {_boundMin[2], _boundMax[2]});
+  }
+  return bounds;
 }
 
 void BoundingBox::expandBy(const BoundingBox &otherBB)
 {
+  PRECICE_ASSERT(_dimensions == otherBB.getDimension(), "Other BoundingBox with different dimensions than this bounding box cannot be used to expand it.");
   for (int d = 0; d < _dimensions; d++) {
-    _bounds[2 * d]     = std::min(_bounds[2 * d], otherBB._bounds[2 * d]);
-    _bounds[2 * d + 1] = std::max(_bounds[2 * d + 1], otherBB._bounds[2 * d + 1]);
+    _boundMin[d] = std::min(_boundMin[d], otherBB._boundMin[d]);
+    _boundMax[d] = std::max(_boundMax[d], otherBB._boundMax[d]);
   }
 }
 
 void BoundingBox::expandBy(const Vertex &vertices)
 {
-  PRECICE_ASSERT(_dimensions == vertices.getDimensions(), "Vertex with different dimensions than bounding box cannot be used to expand bounding box");
+  PRECICE_ASSERT(_dimensions == vertices.getDimensions(), "Vertex with different dimensions than this bounding box cannot be used to expand it.");
   const auto coords = vertices.rawCoords();
   for (int d = 0; d < _dimensions; ++d) {
-    _bounds.at(2 * d)     = std::min(coords[d], _bounds.at(2 * d));
-    _bounds.at(2 * d + 1) = std::max(coords[d], _bounds.at(2 * d + 1));
+    _boundMin[d] = std::min(coords[d], _boundMin[d]);
+    _boundMax[d] = std::max(coords[d], _boundMax[d]);
   }
 }
 
 void BoundingBox::expandBy(double value)
 {
-  if (empty())
-    return;
-  for (int d = 0; d < _dimensions; d++) {
-    _bounds[2 * d] -= value;
-    _bounds[2 * d + 1] += value;
+  if (!isDefault()) {
```

Comment:
What happens if we use here an `PRECICE_ASSERT` instead of a conditional action?

Reply:
There is some logic in the testing scheme that we haven't got the time to look into it deeply, sorry for that. But the testing fails when we remove the conditional with `PRECICE_ASSERT`.